nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -148,11 +152,23 @@ def __init__(self, regressor: str, dimension: int):
             )        
             # And, using partial, we get rid of the parameters of the decision tree (we work on the neural net, not
             # on the decision tree).
-            super().__init__(partial(self._ml_parametrization, dimension=dimension, noise_free=False,
+            super().__init__(partial(self._ml_parametrization, data_dimension=data_dimension, noise_free=False,
                                      regressor="mlp", depth=-3, criterion="no", min_samples_split=0.1), parametrization)
-            self.evaluation_function = partial(self._ml_parametrization, dimension=dimension,  # type: ignore
+            self.evaluation_function = partial(self._ml_parametrization, data_dimension=data_dimension,  # type: ignore
                                                regressor="mlp", noise_free=True, 
                                                depth=-3, criterion="no", min_samples_split=0.1)
         else:
             assert False, f"Problem type {regressor} undefined!"
-        self.register_initialization(regressor=regressor, dimension=dimension)
+        self.register_initialization(regressor=regressor, data_dimension=data_dimension)
```

Comment:
the descriptors are probably missing here

Reply:
done.